stronghold/stronghold.gd

```diff
@@ -1,44 +1,82 @@
 class_name Stronghold
 extends Node2D
 
+@onready var occupants: Node = $Occupants
+@onready var sprites: Node2D = $Sprites
+
 var _health: Health
+var _level: int = 0
+var _max_level: int = 2
+var _base_health: int = 100
+var _current_occupants: int = 0
+var _occupant_slot_progression: Array[int] = [5, 12, 21]
 
 
 func _ready() -> void:
-	GunmanUpgrade.new().level_changed.connect(_handle_gunman_upgrade)
-	_health = Health.new(100)
+	_health = Health.new(_base_health)
 	_health.health_changed.connect(SignalBus.player_health_changed.emit)
 	_health.died.connect(SignalBus.player_died.emit)
 
+	GunmanUpgrade.new().level_changed.connect(_handle_gunman_upgrade)
+	StrongholdUpgrade.new().level_changed.connect(_handle_stronghold_upgrade)
```

Comment:
I think I passed `self` as a parameter to some upgrades, which might be a bit nicer if we want to be limit the global signals.

Reply:
oh yeah, i'll try that out